zipline/finance/commission.py

```diff
@@ -108,3 +111,135 @@ def calculate(self, transaction):
+        """
+        self.equity_cost = float(equity_cost)
+
+        self.equity_min_trade_cost = None if equity_min_trade_cost is None\
+            else float(equity_min_trade_cost)
+
+        if option_costs is not None:
+            assert isinstance(option_costs, dict)
+        self.option_costs = option_costs
+        self.option_min_trade_cost = None if option_min_trade_cost is None\
+            else float(option_min_trade_cost)
+
+        # check that all sids are integers
+        assert isinstance(equity_sids, list)
+        assert isinstance(option_sids, list)
+        for sid in equity_sids:
+            assert(isinstance(sid, int))
+        for sid in option_sids:
+            assert(isinstance(sid, int))
+
+        # sort for later searching with bisect
+        self.equity_sids = sorted(equity_sids)
+        self.option_sids = sorted(option_sids)
+
+    def __repr__(self):
+        return "{class_name}(cost={cost}, min trade cost={min_trade_cost})"\
+            .format(class_name=self.__class__.__name__,
+                    cost=self.cost,
+                    min_trade_cost=self.min_trade_cost)
+
+    def calculate(self, transaction):
+        """
+        returns a tuple of:
+        (per share commission, total transaction commission)
+        """
+        assert isinstance(transaction.sid, int)
+        sid = transaction.sid
+
+        if _list_index(self.equity_sids, sid) is not None:
+            cost = self.equity_cost
+            min_trade_cost = self.equity_min_trade_cost
+        elif _list_index(self.option_sids, sid) is not None:
+            cost = _lookup_option_cost(self.option_costs, transaction.price)
+            min_trade_cost = self.option_min_trade_cost
+        else:
+            raise Exception('Unable to find sid = %i in either the'
+                            ' equities or options.' % sid)
+
+        commission = abs(transaction.amount * cost)
+        if min_trade_cost is None:
+            return cost, commission
+        else:
+            commission = max(commission, min_trade_cost)
+            return abs(commission / transaction.amount), commission
+
+    @staticmethod
+    def ib_default_tier():
+        """
+        Interactive Brokers option commissions tier for < 10,000
+        monthly contracts traded.
+            https://gdcdyn.interactivebrokers.com/en/index.php?f=commission&p=options1
+        """
+        tier_dict = {}
+        tier_dict[(0.10, np.inf)] = 0.70
+        tier_dict[(0.05, 0.10)] = 0.50
+        tier_dict[(-np.inf, 0.05)] = 0.25
+        return tier_dict
+
+
+def _lookup_option_cost(tier_dict, premium):
+    """
+    Returns the per share option commission for a given premium.
+    """
+    if tier_dict is None:
+        return 0.0
+    for (min_val, max_val) in tier_dict:
+        if min_val <= premium < max_val:
+            return tier_dict[(min_val, max_val)]
+    raise Exception('Failed to lookup tiered option cost.')
```

Comment:
This should either be some more special error or just a `KeyError`. `Exception` is too general and hard to catch. We should also include the failing cost in the error message to help users debug the failure.